File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/repository/SyncRepository.kt
package com.scribblefit.feature.ai.domain.repository

import com.scribblefit.feature.ai.domain.model.ParsedWorkout
import com.scribblefit.feature.ai.domain.model.SyncItem
import com.scribblefit.feature.ai.domain.model.SyncStatus
import kotlinx.coroutines.flow.Flow

interface SyncRepository {
    fun getPendingSyncItems(): Flow<List<SyncItem>>
    suspend fun updateSyncStatus(id: String, status: SyncStatus)
    suspend fun saveParsedWorkout(syncItemId: String, workout: ParsedWorkout)
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Domain/Repository/SyncRepository.swift
import Foundation

public enum AISyncStatus: String, Sendable {
    case pending = "PENDING"
    case processing = "PROCESSING"
    case completed = "COMPLETED"
    case failed = "FAILED"
}

public struct AISyncItem: Sendable {
    public let id: String
    public let rawText: String
    public let status: AISyncStatus
    public let createdAt: Date
    
    public init(id: String, rawText: String, status: AISyncStatus, createdAt: Date) {
        self.id = id
        self.rawText = rawText
        self.status = status
        self.createdAt = createdAt
    }
}

public protocol SyncRepository: Sendable {
    func getPendingSyncItems() async throws -> [AISyncItem]
    func updateSyncStatus(id: String, status: AISyncStatus) async throws
    func saveParsedWorkout(syncItemId: String, workout: ParsedWorkout) async throws
    func enqueueScribble(rawText: String) async throws
}
